apps/roam/src/components/CreateRelationDialog.tsx

```diff
@@ -0,0 +1,381 @@
+      uid: targetUid,
+      title: targetTitle,
+      nodes: discourseNodes,
+    });
+    if (selectedTargetType === false) {
+      // should not happen at this point, since the pattern was vetted at input.
+      internalError("Could not find identify node downstream");
+      return null;
+    }
+    const candidateRelations = relDataByTag[selectedRelationName].filter(
+      (rel) => {
+        if (rel.forward) {
+          return (
+            rel.source === selectedSourceType.type &&
+            rel.destination === selectedTargetType.type
+          );
+        } else {
+          return (
+            rel.source === selectedTargetType.type &&
+            rel.destination === selectedSourceType.type
+          );
+        }
+      },
+    );
+    if (candidateRelations.length === 0) {
+      // also should not happen
+      internalError("Could not find the relation");
+      return null;
+    }
+    if (candidateRelations.length !== 1) {
+      // This seems to happen... I need more data.
+      internalError(
+        `Too many relations between ${selectedTargetType.type} and ${selectedSourceType.type}: ${candidateRelations.map((r) => r.id).join(",")}`,
+      );
+      return null;
+    }
+    return candidateRelations[0];
+  };
+
+  const onCreate = async (): Promise<boolean> => {
+    if (selectedTargetUid === undefined) return false;
+    const relation = identifyRelationMatch(
+      selectedTargetTitle,
+      selectedTargetUid,
+    );
+    if (relation === null) return false;
+    const result = await createReifiedRelation({
+      relationBlockUid: relation.id,
+      sourceUid: relation.forward ? sourceNodeUid : selectedTargetUid,
+      destinationUid: relation.forward ? selectedTargetUid : sourceNodeUid,
+    });
+    return result !== undefined;
+  };
+
+  const onCreateSync = (): void => {
+    onCreate()
+      .then((result: boolean) => {
+        if (result) {
+          renderToast({
+            id: `discourse-relation-created-${Date.now()}`,
+            intent: "success",
+            timeout: 10000,
+            content: <span>Created relation</span>,
+          });
+        } else {
+          renderToast({
+            id: `discourse-relation-error-${Date.now()}`,
+            intent: "warning",
+            content: <span>Failed to create relation</span>,
+          });
+        }
+        onClose();
+      })
+      .catch((error) => {
+        renderToast({
+          id: `discourse-relation-error-${Date.now()}`,
+          intent: "danger",
+          content: <span>Failed to create relation</span>,
+        });
+        onClose();
+      });
+  };
+
+  const changeRelationType = (relName: string): void => {
+    setSelectedRelationName(relName);
+    setPageOptions(getFilteredPageNames(relName));
+    if (
+      selectedTargetUid !== undefined &&
+      identifyRelationMatch(selectedTargetTitle, selectedTargetUid) === null
+    ) {
+      setSelectedTargetUid(undefined);
```

Comment:
The text does not clear if the relation is changed to an unsupported relation
https://www.loom.com/share/702625c392e74e699ebef2a07ae281f1